### PR TITLE
API specifications for Case and Evidence

### PR DESCRIPTION
**Pull request by subhashini-egov**
First cut of API specs for Case and Evidence. 

## Summary by CodeRabbit

- **Documentation**
	- Added sequence diagrams to illustrate the processes of registering an advocate, updating an Advocate's details, and registering a litigant.
- **New Features**
	- Enhanced system documentation to provide clearer insights into the workflow for end-users and developers.

### CODE REVIEW

File: api_specifications/evidence-api-1.0.0.yaml [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+      requestBody:
+        description: Details for the new case(s) + RequestInfo meta data.
+        content:
+          '*/*':
+            schema:
+              $ref: >-
+                https://raw.githubusercontent.com/egovernments/DIGIT-OSS/master/core-services/docs/common-contract_v1-1.yml#/components/schemas/EvidenceRequest
+        required: true
+      responses:
+        '201':
```
Comment (atulgupta2024) [inline api_specifications/evidence-api-1.0.0.yaml:55]: typically 201 is only returned when create is done. for all other operations, 200 should be the success code

don't we need a search API as well?

File: api_specifications/evidence-api-1.0.0.yaml [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+        evidenceNumber:
+          type: string
+          description: Formatted evidence number assigned after evidence is accepted by the deciding authority.
+          minLength: 2
+          maxLength: 64
+        externalRefNumber:
+          type: string
+          description: Reference to a physical artifact or other evidence collected during the course of an investigation
+          minLength: 2
+          maxLength: 128
```
Comment (atulgupta2024) [inline api_specifications/evidence-api-1.0.0.yaml:102]: external reference may be longer string. we don't know. suggest we make this 256

**Review by atulgupta2024 — COMMENTED** [review]
comments added

**Review by atulgupta2024 — APPROVED** [review]
the path to AdvocateRequest object is incorrect in places 

https://raw.githubusercontent.com/egovernments/DIGIT-OSS/master/core-services/docs/common-contract_v1-1.yml#/components/schemas/AdvocateRequest

fixed it in my branch and committed.

File: api_specifications/evidence-api-1.0.0.yaml (hunk 1 above)
Comment (subhashini-egov) [reply api_specifications/evidence-api-1.0.0.yaml:55]: Yes, we need search as well. Will add it later.

**Pull request merged**